src/sidebar/services/frame-sync.js

```diff
@@ -165,34 +175,36 @@ export class FrameSyncService {
    */
   _setupSyncFromGuests() {
     // A new annotation, note or highlight was created in the frame
-    this._guestRPC.on('createAnnotation', event => {
-      const annot = Object.assign({}, event.msg, { $tag: event.tag });
-      // If user is not logged in, we can't really create a meaningful highlight
-      // or annotation. Instead, we need to open the sidebar, show an error,
-      // and delete the (unsaved) annotation so it gets un-selected in the
-      // target document
-      if (!this._store.isLoggedIn()) {
-        this._hostRPC.call('openSidebar');
-        this._store.openSidebarPanel('loginPrompt');
-        this._guestRPC.call('deleteAnnotation', formatAnnot(annot));
-        return;
-      }
-      this._inFrame.add(event.tag);
+    this._guestRPC.on(
+      'createAnnotation',
+      (/** @type {AnnotationSafeData} */ annot) => {
```

Comment:
I think you'll need to use an `@param {Type} paramName` annotation _above_ the closure here for TS to use it. See for example uses of `@param` in various `createSelector` calls in store modules.

Reply:
Yes, you are correct. I made those changes in this PR: https://github.com/hypothesis/client/pull/3983